pilot/pkg/bootstrap/server.go

```diff
@@ -960,7 +959,7 @@ func (s *Server) initIstiodCerts(args *PilotArgs, host string) error {
 
 // createPeerCertVerifier creates a SPIFFE certificate verifier with the current istiod configuration.
 func (s *Server) createPeerCertVerifier(tlsOptions TLSOptions) (*spiffe.PeerCertVerifier, error) {
-	if tlsOptions.CaCertFile == "" && s.CA == nil && features.SpiffeBundleEndpoints == "" {
+	if tlsOptions.CaCertFile == "" && s.CA == nil && features.SpiffeBundleEndpoints == "" && !s.DisableCAServer() {
```

Comment:
Why do we need s.CA == nil & s.DisableCAServer()? My understanding is that one of those conditions is sufficient, since s.CA == nil if s.DisableCAServer() is true?

Reply:
we are using `!s.isDisableCa()` so s.CA == nil is needed.